src/commands/deploy.ts

```diff
@@ -47,8 +41,9 @@ export default class Deploy extends Command {
       this.log(`Using options found in ${DEPLOY_OPTIONS_FILE}`);
     }
 
-    let deployers = (await this.config.runHook('project:findDeployers', options)) as Deployer[];
-    deployers = deployers.reduce((x, y) => x.concat(y), [] as Deployer[]);
+    const hookResults = await SfHook.run(this.config, 'sf:deploy', options);
+
+    let deployers = hookResults.successes.map((s) => s.result).reduce((x, y) => x.concat(y), [] as Deployer[]);
```

Comment:
@mdonnalley can reduce be replaced with [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)()?